FrontEnd/src/UserAuthentication/SignUpPage.jsx

```diff
@@ -1,77 +1,124 @@
-import  {useNavigate} from "react-router-dom";
-import {useState} from "react";
-import {registerUser} from "./Auth";
-import {addUser} from './FirestoreDB';
-import './SignUpPage.css';
+import { useNavigate } from "react-router-dom";
+import { useState } from "react";
+import { registerUser } from "./Auth";
+import "./SignUpPage.css";
 
-function SignUpPage(){
-    const navigate = useNavigate();
-    const [email, setEmail] = useState('');
-    const [password, setPassword] = useState('');
-    const [firstName, setFirstName] = useState('');
-    const [lastName, setLastName] = useState('');
-    const [userName, setUserName] = useState('');
-    const MIN_PASSWORD_LENGTH = 6;
-    const handleLoginRoute = () => {
-        navigate(`/LoginPage`);
-      };
+function SignUpPage() {
+  const navigate = useNavigate();
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [userName, setUserName] = useState("");
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+  const MIN_PASSWORD_LENGTH = 6;
+  const handleLoginRoute = () => {
+    navigate(`/LoginPage`);
+  };
 
-    const handleSignUp = async(event) =>{
-        event.preventDefault();
-        if (password.length < MIN_PASSWORD_LENGTH) {
-            alert(`Minimum password length is ${MIN_PASSWORD_LENGTH}`);
-        } else {
-            try{
-                const userCredential = await registerUser(email, password);
-                if (userCredential!==undefined){
-                    addUser(userCredential, firstName, lastName, userName, email);
-                    navigate('/MainPage');
-                } else{
-                    alert("Incorrect Email address");
-                };
-            } catch (error){
-                console.error("Error registering user:", error);
-            };
-        };
-    };
-    return(
+  const handleSignUp = async (event) => {
+    event.preventDefault();
+    if (password.length < MIN_PASSWORD_LENGTH) {
```

Comment:
You can use `minlength` attribute on the `<input />` to enforce this: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/minlength

Reply:
When I use minLength, it doesn't notify users when they are below the minimum. And this creates an error in firebase when password is less than 6 characters but the users won't be able to know without the alert.